flash/data/data_module.py

```diff
@@ -98,44 +98,39 @@ def viz(self) -> BaseViz:
     def viz(self, viz: BaseViz) -> None:
         self._viz = viz
 
-    @classmethod
-    def configure_vis(cls) -> BaseViz:
+    @staticmethod
+    def configure_vis(*args, **kwargs) -> BaseViz:
         return BaseViz()
 
-    def show_batch(self):
-        # viz needs to be enabled, so it doesn't store profile transforms during training
-        with self.viz.enable():
-            _ = next(iter(self.train_dataloader()))
-            _ = next(iter(self.val_dataloader()))
-            _ = next(iter(self.test_dataloader()))
-            _ = next(iter(self.predict_dataloader()))
-
-    def visualize(self, batch: Dict[str, Any], stage: RunningStage) -> None:
+    def show(self, batch: Dict[str, Any], stage: RunningStage) -> None:
```

Comment:
just to leave here as a comment what I suggested in the call: This method and all the other methods callings that one just collect or fetch a batch. I would suggest later to rename to `fetch`. Visualization is just one use case once you have cached the data. A general purpose thing could be profiling or data inspection.

Reply:
Let's change in another PR with team approval.